scanner/src/main/java/ca/on/oicr/gsi/runscanner/scanner/processor/dragen/BCLConvert.java

```diff
@@ -0,0 +1,195 @@
+          // 0 = path, 1 = crc32 checksum
+          Path filename = new File(rootDir, manifestLine[0]).toPath();
+
+          // When this is null, it's often for an Undetermined read. We do not care.
+          try {
+            Analysis analysis = bclConvertAnalysis.get(filename);
+            for (AnalysisFile file : analysis.getFiles()) {
+              if (file.getPath().equals(filename)) {
+                file.setChecksum(manifestLine[1]);
+                break;
+              }
+            }
+          } catch (NullPointerException npe) {
+            log.info("Unable to map {} to an Analysis object", filename);
+          }
+        }
+
+      } else {
+        log.info("No Manifest.tsv for {}", rootDir);
+      }
+
+      // Get read counts from Demultiplex Stats file
+      // for gz compression: root/Analysis/#/Data/BCLConvert/fastq/Reports/Demultiplex_Stats.csv
+      // for ora compression:
+      // root/Analysis/#/Data/BCLConvert/ora_fastq/Reports/Demultiplex_Stats.csv
+      File demulitplexStats =
+          new File(rootDir, "Data/BCLConvert/fastq/Reports/Demultiplex_Stats.csv");
+      if (!(demulitplexStats.exists() && demulitplexStats.isFile())) {
+        demulitplexStats =
+            new File(rootDir, "Data/BCLConvert/ora_fastq/Reports/Demultiplex_Stats.csv");
+      }
+
+      if (demulitplexStats.exists() && demulitplexStats.isFile()) {
+        List<String[]> demultiplexLines =
+            Files.readAllLines(demulitplexStats.toPath())
+                .stream()
+                .map(line -> line.split(","))
+                .toList();
+
+        for (String[] demuxLine : demultiplexLines) {
+          if (demuxLine[0].startsWith("Lane")) continue; // skip the column labels
+          // 0 = Lane, 1 = SampleId, 2 = Index, 3= # Reads, 4 = # Perfect Index Reads,
+          // 5 = # One Mismatch Index Reads, 6 = # Two Mismatch Index Reads, 7 = % Reads,
+          // 8 = % Perfect Index Reads, 9 = % One Mismatch Index Reads,
+          // 10 = % Two Mismatch Index Reads
+          Analysis analysis = bclConvertAnalysis.get(demuxLine[1], demuxLine[0], demuxLine[2]);
+          long readCount = Long.parseLong(demuxLine[3]);
+          // Set the same read count for all files in analysis
+          analysis.getFiles().forEach(file -> file.addInfoItem("read_count", readCount));
+          bclConvertAnalysis.put(analysis);
+        }
+      } else {
+        log.info("No Demultiplex_Stats.csv for {}", rootDir);
+      }
+    } else {
+      log.info("No fastq_list.csv for {}", rootDir);
+    }
+
+    // Check against samplesheet for okayness
+    isOk = true;
+    for (JsonNode item : samplesheet.getInfo().get("BCLConvert").get("Data")) {
+      Analysis analysisItem =
+          bclConvertAnalysis.get(
+              item.get("Sample_ID").textValue(),
+              item.get("Lane").textValue(),
+              item.get("Index").textValue(),
+              item.get("Index2").textValue());
```

Comment:
I believe it's possible to do single-index sequencing. How would that affect this call?

Reply:
Ooh, it's yet more complex: from [an Illumina page](https://knowledge.illumina.com/software/on-premises-software/software-on-premises-software-reference_material-list/000007337) talking about running DRAGEN with the first index read is a UMI and the second index read is an index:
```It is best practice to not leave an empty index column in the sample sheet (in this case an empty column for the first index). Some versions of BCL Convert will accept an empty column while others will error.```

Reply:
Are either of these things we will actually do though?

Reply:
We're handling UMI's downstream of bclconvert, @lheisler I don't think any plans on changing that in the short/medium term? Anyway, it does look like depending on the run bases mask, index one may not be present. Will this set `Index` to null or fail?

Yes, we do single index sequencing, so index two (`Index2`) may be missing too. I'll send you an example run over slack.

Reply:
We will continue to manage UMIs downstream for now.  

Andre had built in some support with bcl2fastq to process UMIs upfront ie. generate fastq with UMIs extracted and recorded.  This was very specific to our UMI design, but we never made use of it.   I don't know what the current state of the Illumina tools and whether it can handle our UMI kit.  It would be worthwhile at some point to come investigate.  One early concern was releasing processed data for RUO projects, where they may have other/better approaches to using the UMIs.   As we've moved to move standard assays, i don't know that this is as much of a concern.   

Our current use of UMIs is that they are includesd in all WG library preps, which precedes capture for TarSeq.  (ie they are also in the TarSeq libraries).  For TarSeq they add value, and it is when these libraries are processed that they are extracted and used.  Data is small and this can be done quickly.   For WG libraries, they are just trimmed off, as they don't add as much value and take a long time to process.